test/DynamicExpresso.UnitTest/ConstructorTest.cs

```diff
@@ -179,6 +180,148 @@ public void Array_multi_dimension_constructor()
+		}
+
+		[Test]
+		public void Ctor_NewMyClass_CanStillUseMemberSyntax()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			Assert.AreEqual(
+				new MyClassAdder() { StrProp = string.Empty, MyArr = new[] { 1, 2, 3, 4, 5 }, IntField = int.MinValue },
+				target.Eval<MyClassAdder>("new MyClassAdder() {StrProp = string.Empty, MyArr = new int[] {1, 2, 3, 4, 5}, IntField = int.MinValue }"));
+		}
+
+		[Test]
+		public void Ctor_InvalidInitializerMemberDeclarator()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7	}"));
+			//Start with collection, then do member init
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},StrProp = \"6\" ,7	}"));
+			//Member init first, then attempt a collection init.
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){StrProp = \"6\" ,{ 1, 2, 3, 4, 5},7	}"));
+		}
+
+		[Test]
+		public void Ctor_CannotUseCollectionInitDoesNotImplementIEnumerable()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			Assert.Throws<ParseException>(() => target.Eval<MyClass>("new MyClass(){ 1, 2, 3, 4, 5}"));
+		}
+
+		[Test]
+		public void Ctor_NewListWithItems()
+		{
+			Ctor_NewListGeneric<string, string>("\"1\"", "\"2\"", "\"3\"", "{string.Empty}", "string.Empty", "int.MaxValue.ToString()", "{int.MinValue.ToString()}");
+			Ctor_NewListGeneric<int, int>("1", "2", "3", "int.MinValue", "int.MaxValue", "{int.MinValue}", "{int.MaxValue}");
+			Ctor_NewListGeneric<object, object>("string.Empty", "int.MinValue");
+		}
+
+		[Test]
+		public void Ctor_NewListCantFindAddMethod()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.List<>));
+			Assert.Throws<ParseException>(() => target.Eval<System.Collections.Generic.List<int>>("string.Empty"));
+			Assert.Throws<ParseException>(() => target.Eval<System.Collections.Generic.List<string>>("int.MaxValue"));
+		}
+
+		public void Ctor_NewListGeneric<TList, TObject>(params string[] items)
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.List<>));
+			target.Reference(typeof(TList));
+			target.Reference(typeof(TObject));
+			//Create a random list of values to test.
+			var actual = new System.Collections.Generic.List<TObject>();
+			foreach (var v in items)
+			{
+				actual.Add(target.Eval<TObject>(v.Trim('}', '{')));
+			}
+			for (var min = 0; min < actual.Count; ++min)
+			{
+				for (var count = Math.Min(min, 1); count <= actual.Count - min; ++count)
+				{
```

Comment:
What's the purpose of the two loops? Wouldn't be enough to just concat all the provided items?